editor/src/messages/layout/layout_message_handler.rs

```diff
@@ -48,6 +48,21 @@ impl<F: Fn(&MessageDiscriminant) -> Vec<KeysGroup>> MessageHandler<LayoutMessage
 		use LayoutMessage::*;
 		#[remain::sorted]
 		match message {
+			CommitLayout { layout_target, widget_id } => {
+				let layout = if let Some(layout) = self.layouts.get_mut(layout_target as usize) {
+					layout
+				} else {
+					warn!("CommitLayout was called referencing an invalid layout. `widget_id: {widget_id}`, `layout_target: {layout_target:?}`",);
+					return;
+				};
+
+				if layout.iter_mut().find(|widget| widget.widget_id == widget_id).is_none() {
+					warn!("CommitLayout was called referencing an invalid widget ID, although the layout target was valid. `widget_id: {widget_id}`, `layout_target: {layout_target:?}`",);
+					return;
+				}
+
+				responses.add(DocumentMessage::StartTransaction);
```

Comment:
It appears this is mistakenly adding undo commits when modifying the tool options (e.g. dragging the weight and stroke colour with the line tool at the top).

![tool options](https://github.com/GraphiteEditor/Graphite/assets/78500760/b3531025-47de-4d0c-a87f-485112f989aa)

Reply:
oh, i see. I missed this condition, maybe I need to do something like on_update callback